### PR TITLE
chore!: set the min evidence height for neutron-1 and stride-1

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: NA

Set the min evidence height for neutron-1 and stride-1 in the v14 upgrade handler. 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API, client, or state breaking change (i.e., requires minor or major version bump)
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry in `.changelog` (for details, see [contributing guidelines](../../CONTRIBUTING.md#changelog))
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: app/upgrades/v14/upgrades.go [hunk 1]
```diff
@@ -21,6 +21,11 @@ func CreateUpgradeHandler(
 			return vm, err
 		}
 
+		// Set the minimum height of a valid consumer equivocation evidence
```
Comment (MSalopek) [inline app/upgrades/v14/upgrades.go:24]: Shouldn't this be a part of migration and not in the upgrade handler?

Having it as a migration would also minimize the risk of "somebody just deleted it on accident".

Comment (mpoke) [reply app/upgrades/v14/upgrades.go:24]: It's Gaia specific and not ICS specific. So it should go in the Gaia upgrade handler. It's like setting the value of a SDK module param.

Comment (MSalopek) [reply app/upgrades/v14/upgrades.go:24]: I'm just nitpicking and leaving the comment for future reference.

There's also this thing: https://github.com/cosmos/gaia/blob/main/app/app.go#L367. By the looks of it, you could also have a separate handler that executes after `CreateUpgradeHandler`.

The code here would apply the provided values on every upgrade - not a big deal 👍 .

**Review by MSalopek — APPROVED** [review]
LGTM

**Review by jtremback — COMMENTED** [review]
LGTM

**Review by insumity — APPROVED** [review]

**Pull request merged**